python/cudf/cudf/_libxx/column.pyx

```diff
@@ -0,0 +1,167 @@
+    def null_count(self):
+        return self.null_count()
+
+    cdef size_type null_count(self) except? 0:
+        return self.view().null_count()
+
+    cdef mutable_column_view mutable_view(self) except *:
+        if is_categorical_dtype(self.dtype):
+            data_dtype = self.codes.dtype
+        else:
+            data_dtype = self.dtype
+        cdef type_id tid = np_to_cudf_types[np.dtype(data_dtype)]
+        cdef data_type dtype = data_type(tid)
+        cdef size_type offset = self.offset
+        cdef vector[mutable_column_view] children
+        cdef void* data
+
+        if is_categorical_dtype(self.dtype):
+            data = <void*><uintptr_t>(self.codes.data.ptr)
+        else:
+            data = <void*><uintptr_t>(self.data.ptr)
+            
+        cdef Column child_column
+        if self.children:
+            for child_column in self.children:
+                children.push_back(child_column.mutable_view())
+
+        cdef bitmask_type* mask
+        if self.mask is not None:
+            mask = <bitmask_type*><uintptr_t>(self.mask.ptr)
+        else:
+            mask = NULL
+
+        return mutable_column_view(
+            dtype,
+            self.size,
+            data,
+            mask,
+            UNKNOWN_NULL_COUNT,
+            offset,
+            children)
+
+    cdef column_view view(self) except *:
+        if is_categorical_dtype(self.dtype):
+            data_dtype = self.codes.dtype
+        else:
+            data_dtype = self.dtype
+        cdef type_id tid = np_to_cudf_types[np.dtype(data_dtype)]
+        cdef data_type dtype = data_type(tid)
+        cdef size_type offset = self.offset
+        cdef vector[column_view] children
+        cdef void* data
+
+        if is_categorical_dtype(self.dtype):
+            data = <void*><uintptr_t>(self.codes.data.ptr)
+        else:
+            data = <void*><uintptr_t>(self.data.ptr)
+        
+        cdef Column child_column
+        if self.children:
+            for child_column in self.children:
+                children.push_back(child_column.view())
+
+        cdef bitmask_type* mask
+        if self.mask is not None:
+            mask = <bitmask_type*><uintptr_t>(self.mask.ptr)
+        else:
+            mask = NULL
+
+        return column_view(
+            dtype,
+            self.size,
+            data,
+            mask,
+            UNKNOWN_NULL_COUNT,
+            offset,
+            children)
+
+    @staticmethod
+    cdef Column from_ptr(unique_ptr[column] c_col):
+        from cudf.core.column import build_column
+
+        size = c_col.get()[0].size()
+        dtype = cudf_to_np_types[c_col.get()[0].type().id()]
+        has_nulls = c_col.get()[0].has_nulls()
+
+        # After call to release(), c_col is unusable
+        cdef column_contents contents = c_col.get()[0].release()
+        
+        data = DeviceBuffer.c_from_unique_ptr(move(contents.data))
+        data = Buffer(data)
+        
+        if has_nulls:
```

Comment:
Just to confirm, say the null_count is 0 so this returns False, but there is a buffer allocated for the mask but it's filled with 1s set because everything is valid. RAII will take care of freeing it appropriately, right?

Reply:
My thought is that since we don’t take ownership of the unique_ptr to the mask. So when contents goes out of scope, so does the unique_ptr, and the memory is freed